parser/lexer.go

```diff
@@ -89,7 +87,10 @@ func (s *Scanner) Errors() (warns []error, errs []error) {
 
 // reset resets the sql string to be scanned.
 func (s *Scanner) reset(sql string) {
-	s.r = reader{s: sql, p: Pos{Line: 1}}
+	if s.r.encoding == nil {
+		s.r.encoding = charset.FindEncoding(mysql.DefaultCharset)
+	}
```

Comment:
the function is `reset`, why we set encoding only if it is `nil`

Reply:
Because we didn't init `r.encoding` when create Scanner. Another method is to set encoding in peek() function. Which one do you prefer?